src/main/java/com/monitorjbl/plugins/MergeBlocker.java

```diff
@@ -37,8 +39,26 @@ public void check(@Nonnull MergeRequest mergeRequest) {
         Set<String> missing = approval.missingRevieiwersNames(pr);
         mergeRequest.veto("Required reviewers must approve", (config.getRequiredReviews() - approval.seenReviewers(pr).size()) +
             " more approvals required from the following users: " + Joiner.on(", ").join(missing));
+      } else {
+        final Boolean blockAutoMergeBecausePrNeedsWork = config.getBlockMergeIfPrNeedsWork() && needsWork(pr);
+
+        if (blockAutoMergeBecausePrNeedsWork) {
+          mergeRequest.veto("PR marked as Needs Work from reviewer(s)", "Resolve it before merge.");
+        }
       }
     }
   }
 
+  private boolean needsWork(final PullRequest pr) {
```

Comment:
Well, current version is perfectly aligned with Bitbucket UI: `Pull request needs work`.